Added Graph class in utils in the file graph.py

Srujan-D opened the pull request.

Srujan-D committed:
The updated common.py file

Srujan-D committed:
Delete common.py

Srujan-D committed:
Merge branch 'restructure-base' of https://www.github.com/ERC-BPGC/gennav

Srujan-D committed:
Merge branch 'master' of https://www.github.com/ERC-BPGC/gennav
Could not see some changes in local repo

Srujan-D committed:
Merge branch 'master' of https://www.github.com/ERC-BPGC/gennav

Srujan-D committed:
Updated Node class in utils.common
1) The argument is changed to **data (Arbitrary keyword arguments (**kwargs)).
2) Keys 'state', 'parent' and 'cost' are included in __init__.
3) The keys have to be specified accordingly when the method is called.

Srujan-D committed:
Added data attribute in Node class to store **data

Srujan-D committed:
Create graph.py

1) Created a Graph class for graph based algorithms.

Srujan-D committed:
Merge branch 'graph_base' into sruj_graph_base

threewisemonkeys-as commented on gennav/utils/graph.py:
```diff
@@ -0,0 +1,39 @@
+        self.edges = defaultdict(list)
+        self.distances = {}
+
+    def add_node(self, node=Node(state=RobotState())):
```
Is it neccesarry to have default value for the node argument?

Srujan-D replied:
No, it's not necessary to include `node=Node(state=RobotState())`, just `node=Node()` is fine. If I don't include that, then even a Point could be added to the graph. So, this just puts a constraint on the user if he just decides to use the Graph class directly (for whatever reason). 
I realize that then the argument will become optional, so if you think my imagination is a little too far off, I can remove the default value.

Srujan-D commented:
I am also thinking of adding another method to calculate distance between the 2 nodes.
So, in the method `add_edge`, there won't be an argument of distance, rather I will just call the `calc_distance` method which will return the calculated distance.

Srujan-D committed:
Added method in Graph class
1) Added method to calculate distance between 2 nodes
2) Changed the Graph class accordingly
3) Removed default values for node arguments everywhere.

Srujan-D committed:
Merge branch 'sruj_graph_base' of https://github.com/Srujan-D/gennav into sruj_graph_base
